src/lib.rs

```diff
@@ -142,8 +146,106 @@ impl Transcript {
         self.strobe.meta_ad(&data_len, true);
         self.strobe.prf(dest, false);
     }
+
+    /// Fork the current `Transcript` to construct an RNG whose output is bound
+    /// to the current transcript state as well as prover's secrets.
+    ///
+    /// See the `TranscriptRng` documentation for more details.
+    pub fn fork_transcript(&self) -> TranscriptRngConstructor {
+        TranscriptRngConstructor {
+            strobe: self.strobe.clone(),
+        }
+    }
+}
+
+/// The prover can commit secrets or randomness to the
+/// `TranscriptRngConstructor` before finalizing to obtain a
+/// `TranscriptRng` which is a PRF of the entire transcript as well as
+/// the prover's secrets and randomness.
+///
+/// See the `TranscriptRng` documentation for more details.
+pub struct TranscriptRngConstructor {
+    strobe: Strobe128,
+}
+
+impl TranscriptRngConstructor {
+    /// Commit witness data to the transcript, so that the finalized
+    /// `TranscriptRng` is a PRF bound to `witness` as well as all
+    /// other transcript data.
+    ///
+    /// The `label` parameter is metadata about the witness, and is
+    /// also committed to the transcript.
+    pub fn commit_witness_bytes(
+        mut self,
+        label: &[u8],
+        witness: &[u8],
+    ) -> TranscriptRngConstructor {
+        let witness_len = encode_usize(witness.len());
+        self.strobe.meta_ad(label, false);
+        self.strobe.meta_ad(&witness_len, true);
+        self.strobe.key(witness, false);
+
+        self
+    }
+
+    /// Use the supplied external `rng` to rekey the transcript, so
+    /// that the finalized `TranscriptRng` is a PRF bound to
+    /// randomness from the external RNG, as well as all other
+    /// transcript data.
+    pub fn reseed_from_rng<R>(mut self, rng: &mut R) -> TranscriptRng
+    where
+        R: rand::Rng + rand::CryptoRng,
+    {
+        let random_bytes = {
+            let mut bytes = [0u8; 32];
+            rng.fill(&mut bytes);
+            bytes
+        };
+
+        self.strobe.meta_ad(b"rng", false);
+        self.strobe.key(&random_bytes, false);
+
+        TranscriptRng {
+            strobe: self.strobe,
+        }
+    }
+}
+
+pub struct TranscriptRng {
+    strobe: Strobe128,
+}
+
+impl TranscriptRng {
+    pub fn labeled_fill_bytes(&mut self, label: &[u8], dest: &mut [u8]) {
```

Comment:
Should we expose this function, or just have people use the `TranscriptRng` as `impl Rng + CryptoRng`?

Reply:
I feel like people should just be using `Rng + CryptoRng`. I'm not sure who would be manually labelling on every squeeze?

Reply:
Ok cool